### DISCUSSION TITLE
Can you set a specific source url to lookup for specific dependencies?

### DISCUSSION THREAD

**Discussion by Andre601** [renovatebot/renovate#24535] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.96.8

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate should look up a specific set of dependencies on a different URL than what it uses by default.

### What you tried so far.

I'll use this section to actually explain the current problem and context.

My project first of all uses a bare-bones `renovate.json` file with only the `$shema` and `labels` option defined.

Next do I use Maven for my project with a `pom.xml` in the repository's root used to hold some main repository URLs that are passed to the individual `pom.xml` files in the different modules (i.e. `https://jitpack.io` is defined in the root pom but used in the module ones).

The problem here seems to be that Renovate doesn't recognize that repositories - alongside dependencies - can be passed down from root pom to module pom, which results in it being unable to look up specific dependencies, reporting failed attempts. As an example one of the warnings is `Failed to look up maven package ch.andre601.advancedserverlist:parent` which is caused by the package not being on maven but jitpack, which is only defined in the root but not the module pom itself.

Basically, I now ask if I can configure Renovate to "tell" it to look up specific URLs for specific dependency types (like the above mentioned), so that this issue is avoided.

I obviously could define the repository in the module pom, but I think that this would increase the maintenance work, as I would need to keep track of and update multiple locations of repository setups instead of one single file, and having this overriden in the config of renovate seems like a more elegant solution.

### Relevant debug logs

<details><summary>Logs</summary>

```
No necessary logs to display here
```

</details>

**Comment by rarkins** [comment, marked as answer]
Yes, use `packageRules` with `registryUrls`. What you describe is the need to change the _registry_ URL, not the _source_ URL.

**Reply by Andre601** [reply]
Alright, thank you.

One other small question. Renovate seems to also check two dependencies that are actually part of my project (I use modules in other modules which renovate seems to see as actual dependencies).

I now want Renovate to ignore those.
I found the `ignoreDeps` option, but are not sure about the values to use here... `matchPackageNames` uses `<groupId:artifactId>` for maven dependencies. Is the same pattern used in `ignoreDeps`?

**Reply by Andre601** [reply]
Also, it seems the docs don't list/mention `registryUrls` as an option for `packageRules`? Maybe that should be corrected if it is indeed missing.

I'm looking at this page btw: https://docs.renovatebot.com/configuration-options/#packagerules

**Reply by Andre601** [reply]
> Also, it seems the docs don't list/mention `registryUrls` as an option for `packageRules`? Maybe that should be corrected if it is indeed missing.
> 
> I'm looking at this page btw: https://docs.renovatebot.com/configuration-options/#packagerules

Found it. It's set as its own option outside the packageRules section, which is a bit weird given it mentions and uses a packageRules example... Maybe move it?

**Reply by rarkins** [reply]
Please stick to one topic per discussion and start a separate discussion if it's a new topic
